### PR TITLE
Proposal for extended weaver configuration

### PR DESCRIPTION
**Pull request by tom-englert**
For https://github.com/Fody/Fody/pull/842

### CODE REVIEW

File: pages/mdsource/configuration.source.md [hunk 1]
```diff
@@ -16,14 +16,59 @@ The file format is:
...
+      </Weavers>
+    </WeaverConfiguration>
+  </PropertyGroup>
+```
+The content of this property is the same as described above
+- You can use MSBuild logic to dynamically control the behavior
+- You can add the configuration e.g. just once in the `Directory.build.props` file to 
+  share the same configuration among several projects.
+- To support sharing the configuration among several projects, entries for weavers that are not installed for a particular project are ignored, so you can configure the superset of all weavers installed in all projects.
+- IntelliSens for the configuration is not available here
```
Comment (ltrzesniewski) [inline pages/mdsource/configuration.source.md:61]:
```suggestion
- IntelliSense for the configuration is not available here
```

**Comment by ltrzesniewski** [comment]
This is not part of the spec, but if the MSBuild property has priority over FodyWeavers.xml, shouldn't a warning be emitted if both are supplied?

**Comment by tom-englert** [comment]
It's not really considered a mistake mixing both, I've just added another chapter to describe the outcome.

**Comment by ltrzesniewski** [comment]
> When multiple configurations are provided, the one with the lowest precedence will control the execution order of the weaver, while the one with the higher precedence will override the content of the configuration node. 

Hmm... I don't really like that, I think it's pretty confusing. I don't think we should be merging the configurations.
Do you have an example of where such behavior could be useful?

**Comment by tom-englert** [comment]
It's the logic that was already implemented for solution/project configuration. You could specify some global default, and have only some project specific exceptions.

E.g. You have a large project with many different `FodyWeaver.xml` files per project. 
Now you can simply add something like this in the `directory.build.props`:

```xml
<WeaverConfiguration Condition="'$(Configuration)' == 'Debug'">
  <Weavers>
    <Costura Exclude="*" />
  </Weavers>
</WeaverConfiguration>
```

**Comment by tom-englert** [comment]
It's the "Expert mode" - if you think it's too confusing, just don't use it...

**Comment by ltrzesniewski** [comment]
Sorry, I misunderstood that part. You're effectively _overwriting_ the individual weaver configs at each level, not actually _merging_ their XML, so that's fine.

**Comment by tom-englert** [comment]
Yes, merging the whole XML would be a nightmare ;-)

**Comment by tom-englert** [comment]
@SimonCropp can you merge this, it's the docu for https://github.com/Fody/Fody/pull/842

**Pull request merged**